src/Sushi.php

```diff
@@ -2,90 +2,91 @@
-
-                $instance->migrate();
-
-                touch($cachePath, filemtime($modelPath));
-            },
-            'no-caching-capabilities' => function () use ($instance) {
-                static::setSqliteConnection(':memory:');
-
-                $instance->migrate();
-            },
-        ];
-
-        switch (true) {
-            case file_exists($cachePath) && filemtime($modelPath) === filemtime($cachePath):
-                $states['cache-file-found-and-up-to-date']();
-                break;
-
-            case file_exists($cacheDirectory) && is_writable($cacheDirectory):
-                $states['cache-file-not-found-or-stale']();
-                break;
-
-            default:
-                $states['no-caching-capabilities']();
-                break;
-        }
+        return app(ConnectionFactory::class);
     }
 
-    protected static function setSqliteConnection($database)
+    public static function migrate(): void
     {
-        static::$sushiConnection = app(ConnectionFactory::class)->make([
-            'driver' => 'sqlite',
-            'database' => $database,
-        ]);
+        $firstRow = static::getRows()->first();
+        $model = new static();
+
+        $model->getConnection()->getSchemaBuilder()->create(
+            $model->getTable(),
+            static function (Blueprint $table) use ($model, $firstRow): void {
+                forward_static_call([static::class, 'getSchema'], $table, $model, $firstRow);
+            }
+        );
+
+        static::getRows()->each(static function (array $row): void {
+            static::create($row);
+        });
     }
 
-    public function migrate()
+    public static function getRows(): LazyCollection
     {
-        $rows = $this->rows;
-        $firstRow = $rows[0];
-        $tableName = $this->getTable();
+        return LazyCollection::make(static::ROWS);
+    }
+
+    public static function getSchema(Blueprint $table, Model $model, array $row): void
+    {
+        if ($model->getIncrementing()) {
+            $table->bigIncrements($model->getKeyName());
+        }
 
-        throw_unless($rows, new \Exception('Sushi: $rows property not found on model: '.get_class($this)));
+        if ($model->usesTimestamps()) {
+            $table->timestamps();
+        }
 
-        static::resolveConnection()->getSchemaBuilder()->create($tableName, function ($table) use ($firstRow) {
-            foreach ($firstRow as $column => $value) {
-                if ($column === 'id') {
-                    $table->increments('id');
-                    continue;
-                }
+        if (method_exists(static::class, 'tapSchema')) {
+            forward_static_call([static::class, 'tapSchema'], $table, $model, $row);
 
-                $type = is_numeric($value) ? 'integer' : 'string';
+            return;
+        }
 
-                $table->{$type}($column);
+        foreach (Arr::except($row, ['id', 'created_at', 'updated_at']) as $column => $value) {
```

Comment:
What if the model rely on different fields for the dates ?
This should not rely on hardcoded names

Reply:
These three are hardcoded because they are handled in the first two `if`s. 
```php
        if ($model->getIncrementing()) {
            $table->bigIncrements($model->getKeyName());
        }

        if ($model->usesTimestamps()) {
            $table->timestamps();
        }
```